api/src/main/java/edu/cornell/mannlib/vitro/webapp/visualization/personpubcount/PersonPublicationCountQueryRunner.java

```diff
@@ -159,27 +162,32 @@ private String getSparqlQuery(String queryURI) {
 	}
 
 	private static class PersonPublicationConsumer extends ResultSetConsumer {
-		Set<Activity> authorDocuments = new HashSet<Activity>();
+		Map<String, Activity> authorDocuments = new HashMap<>();
 
 		@Override
 		protected void processQuerySolution(QuerySolution qs) {
-			Activity biboDocument = new Activity(qs.get("document").asResource().getURI());
+			String documentUri = qs.get("document").asResource().getURI();
+			if (authorDocuments.containsKey(documentUri)) {
 
-			RDFNode publicationDateNode = qs.get("publicationDate");
-			if (publicationDateNode != null) {
-				biboDocument.setActivityDate(publicationDateNode.asLiteral().getString());
-			}
+			} else {
+				Activity biboDocument = new Activity(documentUri);
 
-			RDFNode publicationType = qs.get("publicationType");
-			if (publicationType != null) {
-				biboDocument.setActivityType(publicationType.asResource().getURI());
-			}
+				RDFNode publicationDateNode = qs.get("publicationDate");
+				if (publicationDateNode != null) {
+					biboDocument.setActivityDate(publicationDateNode.asLiteral().getString());
+				}
 
-			authorDocuments.add(biboDocument);
+				RDFNode publicationType = qs.get("publicationType");
+				if (publicationType != null) {
+					biboDocument.addActivityType(publicationType.asResource().getURI());
```

Comment:
Yes. Theoretically, a publication could resolve to having multiple publication types here - although it would normally be considered inadvisable for data to be in that state.

However, as this part of the code can't make a decision as to what the calling code might want to see as a priority for a single publication type, this now records all that it finds, and lets the calling code decide what it wants to do with it.